File: tree/python/b17073.py
import sys
N, W = map(int, input().split())

nodes = [0] * (500001)

for i in range(N - 1):
    node1, node2 = map(int, sys.stdin.readline().split())
    nodes[node1] += 1
    nodes[node2] += 1


leaf_cnt = 0
for i in range(1, N + 1):
    if nodes[i] == 1:
        leaf_cnt += 1

result = W / leaf_cnt
print(round(result, 10))

File: tree/cpp/b17073.cpp
#include <iostream>
using namespace std;

int n, w;
int node[500001]{ 0 };

int main() {

	int node1, node2;
	double leaf = 0;
	double result;
	std::cin >> n >> w;

	for (int i = 0; i < n - 1; i++) {
		std::cin >> node1 >> node2;
		node[node1]++;
		node[node2]++;
	}

	for (int i = 2; i <= n + 1; i++) {
		if (node[i] == 1)
			leaf += 1;
	}

	result = w / leaf;

	std::cout.precision(6);
	std::cout << std::fixed << result << std::endl;

}